src/workflows/airqo_etl_utils/airqo_utils.py

```
@@ -485,31 +485,44 @@ def clean_low_cost_sensor_data(
+                AirQoGxExpectations.from_pandas().gaseous_low_cost_sensor_raw_data_check(
+                    data
+                )
+            case DeviceCategory.LOW_COST:
+                AirQoGxExpectations.from_pandas().pm2_5_low_cost_sensor_raw_data(data)
+        try:
+            data.dropna(subset=["timestamp"], inplace=True)
             data["timestamp"] = pd.to_datetime(data["timestamp"])
-        data.dropna(subset=["timestamp"], inplace=True)
+        except Exception as e:
+            logger.exception(f"There is an issue with the timestamp column: {e}")
+            raise KeyError(f"An error has occurred with the 'timestamp' column: {e}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)